spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteManifestsSparkAction.java

```diff
@@ -261,7 +261,7 @@ private List<ManifestFile> findMatchingManifests() {
       return ImmutableList.of();
     }
 
-    return currentSnapshot.dataManifests().stream()
+    return currentSnapshot.dataManifests(fileIO).stream()
```

Comment:
For this usage, there's a class level `fileIO` that is wrapped to avoid Kryo serialization issues. So we use that instead of `table..io()`.